### PR TITLE
[test] New OpenACC/CUDA/C++ test from MCH

### LINKED ISSUE #248: Add OpenACC + CPP(stdlib) + CUDA test of MCH
MCH will provide the test case.

### LINKED ISSUE #249: Add OpenACC+ MPI + CPP(stdlib) + MPI + CUDA regression test of MCH
MCH will provide the test case.

### PR DESCRIPTION
**Pull request by ajocksch**
This PR also provides a small correction of `automatic_arrays.py`.

Closes #248 
Closes #249

### CODE REVIEW

**Comment by vkarak** [comment]
@ajocksch Please associate this PR with the issue it solves.

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                'MV2_USE_CUDA': '1',
+                'G2G': '1'
+            }
+            self.num_tasks = 8
+            self.num_tasks_per_node = 8
+            self.num_gpus_per_node = 8
+            self._nvidia_sm = '37'
+
+        if withmpi == 'mpi':
+            self.mpiflag = ' -DUSEMPI'
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: Thank you!

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.parameterized_test(['mpi'], ['nompi'])
+class OpenaccCudaMpiCppstd(rfm.RegressionTest):
+    def __init__(self, withmpi):
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: done

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        super().__init__()
+        self.descr = 'test for OpenACC, CUDA, MPI, and C++'
+        self.valid_systems = ['daint:gpu', 'dom:gpu', 'kesch:cn']
+        self.valid_prog_environs = ['PrgEnv-cray', 'PrgEnv-pgi']
+        if self.current_system.name in ['daint', 'dom']:
+            self.modules = ['craype-accel-nvidia60']
+            self._pgi_flags = '-O2 -acc -ta=tesla:cc60 -Mnorpath -lstdc++'
```
[the remark this answers is not in the document]
Comment (teojgo) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: It throws link error `std_cpp_call.o:(.eh_frame+0x12): undefined reference to `__gxx_personality_v0' if the options is not there.

[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: this for some reason required

std_cpp_call.o:(.eh_frame+0x12): undefined reference to `__gxx_personality_v0'

I could also try to move it into the makefile

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.parameterized_test(['mpi'], ['nompi'])
+class OpenaccCudaMpiCppstd(rfm.RegressionTest):
```
[the remark this answers is not in the document]
Comment (ajocksch) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: done

**Comment by ajocksch** [comment]
currently the check with pgi+mpi fails, the reeason should be the wrong PrgEnv modules; this should be fixed when merging into the master

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (vkarak) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: It's fine here. No problem.

**Comment by ajocksch** [comment]
about 1.: Do you mean to compile with -hnoacc? I think this is not possible since a cuda kernel is called, we would need to provide a cpu version of that part of the code as well.

**Comment by teojgo** [comment]
@jenkins-cscs retry all

**Comment by teojgo** [comment]
lgtm. @ajocksch are the kesch failures expected?

**Comment by ajocksch** [comment]
#348 needs to be merged in the master first

File: cscs-checks/mch/openacc_cuda_mpi_cppstd.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+@rfm.parameterized_test([True], [False])
+class OpenaccCudaMpiNoMPICppstd(rfm.RegressionTest):
```
Comment (vkarak) [inline cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: The name is weird! Why not just `OpenaccCudaCpp`, which is in accordance with the issue as well?

Comment (vkarak) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: @teojgo Can you change the name of this class? Other than this, this is ready to be merged.

Comment (vkarak) [reply cscs-checks/mch/openacc_cuda_mpi_cppstd.py]: I don't think the automatically generated name will be very meaningful for this one: `OpenaccCudaCpp_False` and `OpenaccCudaCpp_True`. So it should be set manually.

**Comment by vkarak** [comment]
@ajocksch Can you check if the failures on Kesch are expected? IMO, perhaps they are not. If they are, though, I can merge this immediately.

**Comment by vkarak** [comment]
@jenkins-cscs retry dom

**Pull request merged**